hnn_core/pyramidal.py

```diff
@@ -1046,25 +975,18 @@ def parreceive_ext(self, type, gid, gid_dict, pos_dict, p_ext):
                     'type_src': type
                 }
 
-                self.ncfrom_extpois.append(
-                    self.parconnect_from_src(
-                        gid_extpois, nc_dict, self.basal2_ampa))
-                self.ncfrom_extpois.append(
-                    self.parconnect_from_src(
-                        gid_extpois, nc_dict, self.basal3_ampa))
-                self.ncfrom_extpois.append(
-                    self.parconnect_from_src(
-                        gid_extpois, nc_dict, self.apicaloblique_ampa))
+                self._connect_at_loc(
+                    loc='proximal', receptor='ampa',
+                    gid_src=gid_extpois,
+                    nc_dict=nc_dict,
+                    nc_list=self.ncfrom_extpois)
 
                 if p_ext[self.celltype][1] > 0.0:
```

Comment:
this if condition is awkward but I didn't want to break things, so left it there

Reply:
Agreed. We'll need to fix this when we remove the loops that iterate through the different feed types from the cell classes. I'm not sure why this was put here in the first place (maybe network build time?) since an nmda receptor conductance of of zero shouldn't have any effect on the simulation.

Reply:
isn't this related to your PRs about turning off inputs based on the weights? or was that different?

> maybe network build time?

I hope not. "Premature optimization is the root of all evil." (Knuth) Another related aphorism is [make it work, make it nice, make it fast](https://wiki.c2.com/?MakeItWorkMakeItRightMakeItFast)

Network build times are really small compared to simulation time and developer time (most important in my opinion). Maybe for GUI application, some of these times matter but I'd rather refactor first, then profile and worry about timing.

Reply:
> isn't this related to your PRs about turning off inputs based on the weights? or was that different?

It's related in principle, but I was unaware that there was a check for synaptic weight >0 within a cell class method. There is probably more to the story that we are unaware of, but we'll need to handle this more consistently in the future.